e2e/nomostest/new.go

```diff
@@ -400,7 +394,7 @@ func setupTestCase(nt *NT, opts *ntopts.New) {
 		nt.T.Fatalf("waiting for ConfigSync Deployments to become available: %v", err)
 	}
 
-	nt.PortForwardOtelCollector()
+	nt.portForwardOtelCollector()
```

Comment:
What if test setup just makes a new empty PortForwarder for the otel-collector and git-server when it creates the test NT. Then when you call nt.GitServerPortForwarder.GetPort() it lazily starts forwarding?

Reply:
Usually the way to handle that problem is to have a private method that only the struct calls and then the public methods owns the lock.

Reply:
updated to lazily start forwarding

Reply:
thanks!